tracr/craft/vectorspace_fns.py

```diff
@@ -97,13 +94,18 @@ def combine_in_parallel(cls, fns: Sequence["Linear"]) -> "Linear":
         *[fn.input_space for fn in fns])
     joint_output_space = bases.join_vector_spaces(
         *[fn.output_space for fn in fns])
+    
+    # precompute properties for these parents to avoid recomputing for every single child
```

Comment:
Why is this important? Shouldn't the index be computed the first time it's called and then not anymore?

Reply:
oh this is because it gets computed for all the children and the updates don't propagate up to the parent, so if we precompute it for the parent we avoid making up to thousands of repeats of the same computation for all the children, I'll modify the comment to reflect this

Reply:
Makes sense, thanks!